jsonnet/components/metrics-server.libsonnet

```diff
@@ -0,0 +1,374 @@
+      apiGroup: 'rbac.authorization.k8s.io',
+      kind: 'Role',
+      name: 'extension-apiserver-authentication-reader',
+    },
+    subjects: [
+      {
+        kind: 'ServiceAccount',
+        name: 'metrics-server',
+        namespace: cfg.namespace,
+      },
+    ],
+  },
+  clusterRoleBindingAuthDelegator: {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'ClusterRoleBinding',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'auth-delegator',
+        'app.kubernetes.io/component': 'metrics-server',
+      } + cfg.commonLabels,
+      name: 'metrics-server:system:auth-delegator',
+    },
+    roleRef: {
+      apiGroup: 'rbac.authorization.k8s.io',
+      kind: 'ClusterRole',
+      name: 'system:auth-delegator',
+    },
+    subjects: [
+      {
+        kind: 'ServiceAccount',
+        name: 'metrics-server',
+        namespace: cfg.namespace,
+      },
+    ],
+  },
+  clusterRoleBinding: {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'ClusterRoleBinding',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'metrics-server',
+      } + cfg.commonLabels,
+      name: 'system:metrics-server',
+    },
+    roleRef: {
+      apiGroup: 'rbac.authorization.k8s.io',
+      kind: 'ClusterRole',
+      name: 'system:metrics-server',
+    },
+    subjects: [
+      {
+        kind: 'ServiceAccount',
+        name: 'metrics-server',
+        namespace: cfg.namespace,
+      },
+    ],
+  },
+  service: {
+    apiVersion: 'v1',
+    kind: 'Service',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'metrics-server',
+        'app.kubernetes.io/component': 'metrics-server',
+      } + cfg.commonLabels,
+      annotations: {
+        'service.beta.openshift.io/serving-cert-secret-name': 'metrics-server-tls',
+      },
+      name: 'metrics-server',
+      namespace: cfg.namespace,
+    },
+    spec: {
+      ports: [
+        {
+          name: 'https',
+          port: 443,
+          protocol: 'TCP',
+          targetPort: 'https',
+        },
+      ],
+      selector: {
+        'app.kubernetes.io/name': 'metrics-server',
+      } + cfg.commonLabels,
+    },
+  },
+  deployment: {
+    apiVersion: 'apps/v1',
+    kind: 'Deployment',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'metrics-server',
+        'app.kubernetes.io/component': 'metrics-server',
+      } + cfg.commonLabels,
+      name: 'metrics-server',
+      namespace: cfg.namespace,
+    },
+    spec: {
+      replicas: 2,
+      selector: {
+        matchLabels: {
+          'app.kubernetes.io/name': 'metrics-server',
+        },
+      },
+      strategy: {
+        rollingUpdate: {
+          maxUnavailable: 1,
+        },
+      },
+      template: {
+        metadata: {
+          labels: {
+            'app.kubernetes.io/name': 'metrics-server',
+            'app.kubernetes.io/component': 'metrics-server',
+          } + cfg.commonLabels,
+        },
+        spec: {
```

Comment:
I think that we discussed and agreed on using minReadySeconds to ensure that metrics availability on upgrades?